### DISCUSSION TITLE
Longer Block Effect Range

### DISCUSSION THREAD

**Discussion by LucasBernhard** [OreCruncher/DynamicSurroundingsFabric#108] [Ideas]
HI OreCruncher

I was wondering if it would be possible to have longer block effect ranges ? Or is there engine limitations that prevents it ?
It would be awesome to be able to hear waterfalls from far away

**Comment by OreCruncher** [comment]
For a given sound event (definition in the sounds.json) a distance can be specified.  If not specified, the range is 16 blocks meaning distances greater than 16 blocks from the player will not be heard. Some of the waterfall sound events are configured for distances > 16 blocks.  Unfortunately, this is baked into the json and can't really be configured dynamically.

Other thing is increasing the range of waterfall effects. Imagine an invisible cube that surrounds the player. The effects system has to scan all the blocks within that cube, once.  As the player moves through the world new blocks come into this invisible cube, and others leave. If the size of the cube increases (i.e., increase the range) there will be more blocks processed.  This will affect client side TPS.

I have been making optimizations with the intent of increasing the size of the cube. It's a work in progress.
